### PR TITLE
Add option to sanitize claim file result output

### PR DESCRIPTION
**Pull request by sebrandon1**
Adds a flag `--sanitize-claim` which defaults to false.  This is mostly used for QE.

Modifies the `NewLabelsExprEvaluator` and related functions to its own package named `labels`.

### CODE REVIEW

File: pkg/claimhelper/claimhelper.go [hunk 1]
```diff
@@ -374,3 +376,30 @@ func CreateClaimRoot() *claim.Root {
...
+	if err != nil {
+		log.Error("ReadClaimFile failed with err: %v", err)
+		return "", err
+	}
+	var aRoot claim.Root
+	UnmarshalClaim(data, &aRoot)
+
+	// Remove the results that do not match the labels filter
+	for testID := range aRoot.Claim.Results {
+		if aRoot.Claim.Results[testID].TestID.Id != labelsFilter {
```
Comment (jmontesi) [inline pkg/claimhelper/claimhelper.go:392]: Does this work for a `labelsFilter` that is not the name of a test case? For instance for a test suite name or a label such as `telco` or `all`.

File: pkg/claimhelper/claimhelper.go [hunk 2]
```diff
@@ -374,3 +376,30 @@ func CreateClaimRoot() *claim.Root {
...
+			log.Info("Removing test ID: %s from the claim", testID)
+			delete(aRoot.Claim.Results, testID)
+		}
+	}
+
+	log.Info("Number of results after sanitizing: %d", len(aRoot.Claim.Results))
```
Comment (jmontesi) [inline pkg/claimhelper/claimhelper.go:398]: These log messages seem irrelevant.

File: pkg/claimhelper/claimhelper.go [hunk 3]
```diff
@@ -148,6 +148,11 @@ func (c *ClaimBuilder) Build(outputFile string) {
 	log.Info("Claim file created at %s", outputFile)
 }
 
+func (c *ClaimBuilder) GetClaimResults() *claim.Root {
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go:151]: Ah yes, I was using this for experimenting.  I'll remove.

File: pkg/claimhelper/claimhelper.go (hunk 1 above)
Comment (jmontesi) [reply pkg/claimhelper/claimhelper.go:392]: You could make it work for any `labelsFilter` by using the `labelsExprEvaluator` (and making it available outside the `checksdb` package) like this:

`if !labelsExprEvaluator.Eval(aRoot.Claim.Results[testID].TestID.Id) {
// delete test
}`

Otherwise I think that the functionality is too restricted to make a flag out of it.

Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go]: Done, this works.

File: pkg/claimhelper/claimhelper.go (hunk 2 above)
Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go:400]: Removed.

**Review by edcdavid — COMMENTED** [review]
when using the --sanitize-claim, the parser output is broken. It does not display any tests. Investigating ...
I used a suite label "access-control" and there is not tests under "results"

File: pkg/claimhelper/claimhelper.go (hunk 1 above)
Comment (edcdavid) [reply pkg/claimhelper/claimhelper.go]: I couldn't make this work, looks like for each test ID read from the claim file, we need to generate metadata with identifiers.GetTestIDAndLabels(identifiers.TestSysAdminIdentifier), including the suite name and tag

Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go]: Ah I see, yes I wonder if I change:
`if !evaluator.Eval([]string{aRoot.Claim.Results[testID].TestID.Id})`
to 
`if !evaluator.Eval([]string{aRoot.Claim.Results[testID].TestID.Id, aRoot.Claim.Results[testID].TestID.Suite})`

Since the Eval() func accepts a slice of strings to evaluate.  I'll try that out.

Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go]: Okay yes this worked.  I was able to run:
`./certsuite run --sanitize-claim -l access-control`

And it keeps all of the access-control related tests in the claim output.

Comment (edcdavid) [reply pkg/claimhelper/claimhelper.go]: What about just calling the function to return the labels to match on with something like:
```
		_,labels:=identifiers.GetTestIDAndLabels(*aRoot.Claim.Results[testID].TestID)
		if !evaluator.Eval( labels) {
```
That would also cover tags

Comment (sebrandon1) [reply pkg/claimhelper/claimhelper.go]: This works as well.  Updated!

**Review by edcdavid — APPROVED** [review]
